sdk/apis/tenancy/v1alpha1/types_mounts.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+
+	corev1alpha1 "github.com/kcp-dev/kcp/sdk/apis/core/v1alpha1"
+	conditionsv1alpha1 "github.com/kcp-dev/kcp/sdk/apis/third_party/conditions/apis/conditions/v1alpha1"
+)
+
+// Mount is a workspace mount that can be used to mount a workspace into another workspace or resource.
+// Mounting itself is done at front proxy level.
+type Mount struct {
```

Comment:
@sttts  I think I was not clear on the intention of this. I think this iteration represents a better view:
0. Source is used to trigger reconcile for external components. 
1. Name is used to map to status field (unique per Workspace)
2. Status is a placeholder for status to be put by an external component.

Reply:
you mean that one resource (e.g. KubeCluster) could provider multiple mounts in multiple workspace? Thinking about one `mycluster` workspace and one `mycluster-sudo` workspace.

Reply:
and of course we could default that value to `"default"`.

Reply:
Its possible, hecne list of mounts and statuses
 Lets say "sudo" could be different mountpoint, mountpoints per namespaces" where we mount only 1 namespace could be mountpoint,